docs/v8-updates.md

```diff
@@ -0,0 +1,86 @@
+# V8 Updates
+
+To update the version of V8 used by workerd, the steps are:
+
+1. Check https://omahaproxy.appspot.com/ and identify the latest version of V8 used by the beta versions of Chrome beta.
+
+2. Install depot_tools if it is not already present on your machine.
+
+   https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up
+
+3. Fetch V8
+
+   ```
+   $ mkdir v8
+   $ cd v8
+   $ fetch v8
+   ```
+
+4. Sync V8 to the version of V8 currently used by workerd.
+
+   First find, the commit hash of the "v8" `git_repository` in the workerd [WORKSPACE](../WORKSPACE) file.
+
+   Then sync your fetched version v8 so that it corresponds to that hash.
+
+   ```
+   $ cd v8/v8
+   $ git checkout <commit_hash>
+   $ gclient sync
+   ```
+
+5. Create a V8 branch for workerd's V8 patches in your recently fetch V8.
+
+   ```
+   $ git checkout -b workerd-patches
+   $ git am <path_to_workerd>/patches/v8/*
+   ```
+
+7. Rebase workerd V8 changes onto the new version of V8. For example, assuming
+   we are updating to 11.4.183.8 and there are 8 workerd patches for V8, the
+   command would be:
+
+   ```
+   $ git rebase --onto 11.4.183.8 HEAD~8
+   ```
+
+   There is usually some minor patch editing required during rebase.
+
+   Ideally at this stage, you should be able to build and test V8 with the patches applied.
+
+8. Re-generate workerd's v8 patches. Assuming there are 8 workerd patches for V8,
+   the command would be:
+
+   ```
+   $ git format-patch -k --no-signature HEAD~8
```

Comment:
May I suggest `--full-index`, to keep the full hashes in the patch files? It doesn't really matter either way, so long as everyone does the same thing (to minimize unnecessary diffs), and that's what we've standardized on internally.